script/release/notes/notes.js

```diff
@@ -319,25 +319,14 @@ async function runRetryable (fn, maxRetries) {
 
 const getPullRequest = async (number, owner, repo) => {
   const name = `${owner}-${repo}-pull-${number}`
-  return checkCache(name, async () => {
-    return runRetryable(octokit.pulls.get({
-      number,
-      owner,
-      repo
-    }), MAX_FAIL_COUNT)
-  })
+  const retryableFunc = () => octokit.pulls.get({ pull_number: number, owner, repo })
+  return checkCache(name, async () => runRetryable(retryableFunc, MAX_FAIL_COUNT))
```

Comment:
Since `octokit.pulls.get` returns a promise, we'd want to `await runRetryable(retryableFunc, MAX_FAIL_COUNT))` here, otherwise the `async` keyword wouldn't be needed

Reply:
To put this at the front: you're right, the second param to `checkCache()` doesn't need to be an async function here. It will still work, and be more readable, if we remove the `async` keyword from the second line here..

That said, as an academic exercise I'm wondering why this works in 67311dd. `async() => runRetryable(retryableFunc, MAX_FAIL_COUNT)` is an async function which returns a promise, so `await operation()` in `checkCache()` seems like it should resolve with the inner promise, rather than the octokit response object. So why does the octokit response object get cached instead of a pending promise? :thinking:

Reply:
answer: because if you `await` a Promise has an inner Promise, the `await` will fulfill with the inner Promise's value.

If x is a promise, adopt its state [3.4]:
If x is pending, promise must remain pending until x is fulfilled or rejected.
If/when x is fulfilled, fulfill promise with the same value.

https://promisesaplus.com/#point-49